test/testsyncmove.cpp

```diff
@@ -841,21 +841,73 @@ private slots:
 
     }
 
+    void testRenameParallelism_data()
+    {
+        QTest::addColumn<Vfs::Mode>("vfsMode");
+
+        QTest::newRow("Vfs::Off") << Vfs::Off;
+#ifdef Q_OS_WIN32
+        if (isVfsPluginAvailable(Vfs::WindowsCfApi)) {
+            QTest::newRow("Vfs::WindowsCfApi") << Vfs::WindowsCfApi;
+        } else {
+            QFAIL("Skipping Vfs::WindowsCfApi");
+        }
+#endif
+    }
+
     // Test that deletes don't run before renames
     void testRenameParallelism()
     {
-        FakeFolder fakeFolder{ FileInfo{} };
+        QFETCH(Vfs::Mode, vfsMode);
+
+        FakeFolder fakeFolder({ FileInfo{} }, vfsMode);
+
+        FileInfo::ComparissonOption cmpOpt = FileInfo::ContentIsKing;
+        if (vfsMode != Vfs::Off)
+        {
+            auto vfs = QSharedPointer<Vfs>(createVfsFromPlugin(vfsMode).release());
+            QVERIFY(vfs);
+            fakeFolder.switchToVfs(vfs);
+            fakeFolder.syncJournal().internalPinStates().setForPath("", PinState::OnlineOnly);
+
+            // make files virtual
+            fakeFolder.syncOnce();
+
+            cmpOpt = FileInfo::IgnoreContentOfDehydratedFiles;
+        }
+
         fakeFolder.remoteModifier().mkdir("A");
         fakeFolder.remoteModifier().insert("A/file");
         QVERIFY(fakeFolder.syncOnce());
-        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        {
+            auto localState = fakeFolder.currentLocalState();
+            FileInfo *file = localState.find({ "A/file" });
+            QVERIFY(file != nullptr); // check if the file exists
+            if (vfsMode != Vfs::Off) {
+                QVERIFY(file->isDehydratedPlaceholder);
+            }
+            qDebug()<<"local"<<localState;
+            qDebug()<<"remote:"<<fakeFolder.currentRemoteState();
+            QVERIFY(localState.equals(fakeFolder.currentRemoteState(), cmpOpt));
+        }
 
         fakeFolder.localModifier().mkdir("B");
         fakeFolder.localModifier().rename("A/file", "B/file");
         fakeFolder.localModifier().remove("A");
-
         QVERIFY(fakeFolder.syncOnce());
-        QCOMPARE(fakeFolder.currentLocalState(), fakeFolder.currentRemoteState());
+
+        {
+            auto localState = fakeFolder.currentLocalState();
+            QVERIFY(localState.find("A/file") == nullptr); // check if the file is gone
+            QVERIFY(localState.find("A") == nullptr); // check if the directory is gone
+            FileInfo *file = localState.find({ "B/file" });
+            QVERIFY(file != nullptr); // check if the file exists
+            if (vfsMode != Vfs::Off) {
+                QVERIFY(file->isDehydratedPlaceholder); // check that no-one messed with the placeholder state
+            }
+            QVERIFY(localState.equals(fakeFolder.currentRemoteState(), cmpOpt));
```

Comment:
The benefit of a QCOMPARE here is that we had human readable errors in the log

Reply:
hmm would be awesome if we could wrap all this complexity behind some magic else our tests complexity will increase exponential 🤯

Reply:
I'm wondering if we should get rid of the old `==` implementation. But I don't know how many tests that'd break.